src/training/data.py

```diff
@@ -385,13 +418,30 @@ def get_wds_dataset(args, preprocess_img, is_train, epoch=0, floor=False, tokeni
             # at this point, we have an iterator over the shards assigned to each worker
             wds.tarfile_to_samples(handler=log_and_continue),
         ])
+
+    def txt_filter(txt):
+
+        invalid_chars = ['`', '~', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '=', '+']
+        invalid_chars += ['[', '{', ']', '}', '\\', '|', ';', ':', "'", '"', ',', '<', '.', '>', '/', '?']
+
+        for c in invalid_chars:
+            txt = txt.replace(c, ' ')
+        return txt
+
+    def preprocess_txt(txt):
+
+        if isinstance(txt, list):
+            txt = txt[0]
+
+        return tokenizer(txt_filter(txt).split(' '))
```

Comment:
Is it correct that the tokenizer expects as list of strings? As there is no type annotation, it's hard for me to see its expected arguments.

Reply:
i'm actually not entirely sure, i actually think sometimes it's a list of strings, and sometimes its a single string